src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,114 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final int NEW_SIZE_MULTIPLIER = 2;
+    private Node<K, V>[] table;
+    private int threshold;
+    private int size;
+
+    public MyHashMap() {
+        table = (Node<K, V>[]) new Node[DEFAULT_CAPACITY];
+        threshold = (int) (LOAD_FACTOR * table.length);
+    }
 
     @Override
     public void put(K key, V value) {
+        if (size == threshold) {
+            resize();
+        }
+        Node<K, V> newNode = new Node<>(key, value);
+        int index = getIndex(key);
+        if (table[index] == null) {
+            table[index] = newNode;
+            size++;
+            return;
+        } else {
+            putWithCollision(newNode);
+        }
 
     }
 
     @Override
     public V getValue(K key) {
+        for (Node<K, V> node = table[getIndex(key)]; node != null; node = node.next) {
+            if (getKey(node) == key
+                    || (getKey(node) != null
+                    && getKey(node).equals(key))) {
+                return getValueFromNode(node);
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    public static class Node<K, V> {
```

Comment:
```suggestion
    private static class Node<K, V> {
```

I think we could make this class private